File: codewars/6kyu/stringing-me-along/dduck-stringing-me-along.js
// https://www.codewars.com/kata/55f4a44eb72a0fa91600001e

const createMessage = (s, res = '') => (s ? (newStr) => createMessage(newStr, `${res} ${s}`) : res.slice(1));


const t1 = createMessage('Hello')('World!')('how')('are')('you?')();
const t2 = 'Hello World! how are you?';
console.log(t1 === t2);

File: codewars/6kyu/stringing-me-along/dduck-stringing-me-along.py
# https://www.codewars.com/kata/55f4a44eb72a0fa91600001e

from unittest import TestCase

create_message = lambda s="", res="": (lambda newS="": create_message(newS, f'{res} {s}')) if s else res[1:]

TestCase().assertEqual(create_message("Hello")("World!")("how")("are")("you?")(), "Hello World! how are you?")
